### PR TITLE
Cosmos triggered Functions

### PR DESCRIPTION
**Pull request by mikhailshilkov**
Part of #218

### CODE REVIEW

File: sdk/nodejs/cosmosdb/zMixins.ts [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+    /**
+     * When set, it adds a prefix to the leases created in the Lease collection for this Function, effectively allowing 
+     * two separate Azure Functions to share the same Lease collection by using different prefixes.
+     */
+    leaseCollectionPrefix: pulumi.Input<string>;
+
+    /**
+     * When set to true, the leases collection is automatically created when it doesn't already exist. 
+     * The default value is false.
```
Comment (CyrusNajmabadi) [inline sdk/nodejs/cosmosdb/zMixins.ts]: it says default is false, but passes 'true' below.

File: sdk/nodejs/cosmosdb/zMixins.ts [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        const { resourceGroupName, location } = appservice.getResourceGroupNameAndLocation(args, account.resourceGroupName);
+
+        const bindingConnectionKey = "BindingConnectionAppSettingsKey";
+
+        const bindings: CosmosBindingDefinition[] = [{
+            name: "items",
+            direction: "in",
+            type: "cosmosDBTrigger",
+            leaseCollectionPrefix: name,
+            createLeaseCollectionIfNotExists: true,
```
Comment (CyrusNajmabadi) [inline sdk/nodejs/cosmosdb/zMixins.ts]: should this be user configurable?

File: sdk/nodejs/cosmosdb/zMixins.ts (hunk 1 above)
Comment (mikhailshilkov) [reply sdk/nodejs/cosmosdb/zMixins.ts]: If we don't specify it in binding properties, Azure will assume `false`. That's why we pass `true` below. Too confusing?

File: sdk/nodejs/cosmosdb/zMixins.ts (hunk 2 above)
Comment (mikhailshilkov) [reply sdk/nodejs/cosmosdb/zMixins.ts]: Do we want to expose all properties as configurable? (there are more of them) Or take some defaults for now, which make sense in 95% of cases? (my call)

File: sdk/nodejs/cosmosdb/zMixins.ts (hunk 1 above)
Comment (CyrusNajmabadi) [reply sdk/nodejs/cosmosdb/zMixins.ts]: gotcha.  it is a bit confusing.  so we should just explain what's going on in the code (either here or at the poitn you pass in 'true').  we should also state why it's appropriate to pass in 'true' .

File: sdk/nodejs/cosmosdb/zMixins.ts (hunk 2 above)
Comment (CyrusNajmabadi) [reply sdk/nodejs/cosmosdb/zMixins.ts]: we can totally not expose if we think this is reasonable.  users can always request in the future.  though we should likely doc why we think this is the right default.

Comment (mikhailshilkov) [reply sdk/nodejs/cosmosdb/zMixins.ts]: Added a comment to cover both.

**Comment by henriktao** [comment]
Is onChange() still the recommended way of triggering functions through cosmos DB? I have encountered a problem that the lease collection is not created even though createLeaseCollectionIfNotExists is set to true.

**Pull request merged**